src/plugins/intel_cpu/src/nodes/kernels/gather_uni_kernel.cpp

```diff
@@ -10,1036 +10,1157 @@ using namespace dnnl::impl::cpu;
 namespace ov {
 namespace intel_cpu {
 
-const unsigned jitGatherKernelBase::shufMask8bitUni[16]  = {0x0C080400, 0x80808080, 0x80808080, 0x80808080, 0x0C080400, 0x80808080, 0x80808080, 0x80808080,
-                                                            0x0C080400, 0x80808080, 0x80808080, 0x80808080, 0x0C080400, 0x80808080, 0x80808080, 0x80808080};
-const unsigned jitGatherKernelBase::permMask8bitA2[8]    = {0, 4, 1, 5, 2, 6, 3, 7};
-const unsigned jitGatherKernelBase::permMask8bitA5[16]   = {0, 4, 8, 12, 1, 5, 9, 13, 2, 6, 10, 14, 3, 7, 11, 15};
-
-const unsigned jitGatherKernelBase::shufMask16bitUni[16] = {0x05040100, 0x0D0C0908, 0x80808080, 0x80808080, 0x05040100, 0x0D0C0908, 0x80808080, 0x80808080,
-                                                            0x05040100, 0x0D0C0908, 0x80808080, 0x80808080, 0x05040100, 0x0D0C0908, 0x80808080, 0x80808080};
-const unsigned jitGatherKernelBase::permMask16bitA2[8]   = {0, 1, 4, 5, 2, 3, 6, 7};
-const unsigned jitGatherKernelBase::permMask16bitA5[16]  = {0, 1, 4, 5, 8, 9, 12, 13, 2, 3, 6, 7, 10, 11, 14, 15};
-
-const unsigned jitGatherKernelBase::incVec[16] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15};
-
 #define GET_OFF(field) offsetof(gatherJitExecArgs, field)
 
-template <x64::cpu_isa_t isa>
-jitUniGatherKernel<isa>::jitUniGatherKernel(const jGatherConfParams& jcp) :
-        jitGatherKernelBase(jcp), x64::jit_generator() {
-    vlen = x64::cpu_isa_traits<isa>::vlen;
-    dataElPerVec = vlen / jcp.dataTypeSize;
-    idxElPerVec = vlen / indicesTypeSize;
-    if (jcp.dataTypeSize == 2)
-        dataTypeShift = 1;
-    else if (jcp.dataTypeSize == 4)
-        dataTypeShift = 2;
 
-    if (isa == x64::avx2) {
-        permMask8bitUni = permMask8bitA2;
-        permMask16bitUni = permMask16bitA2;
-    } else if (isa == x64::avx512_core) {
-        permMask8bitUni = permMask8bitA5;
-        permMask16bitUni = permMask16bitA5;
-    }
+template<x64::cpu_isa_t isa>
+void jitGatherKernelBase<isa>::initialize(const jGatherConfParams& jcp) {
+    this->beforeAxisSize = jcp.beforeAxisSize;
+    this->specIdxSize = jcp.specIdxSize;
+    this->batchDims = jcp.batchDims;
+    this->reverseIndexing = jcp.reverseIndexing;
+    this->afterAxisSize = jcp.afterAxisSize;
+    this->dynamicShapes = jcp.dynamicShapes;
 }
 
-template <x64::cpu_isa_t isa>
-void jitUniGatherKernel<isa>::create_ker() {
+template<x64::cpu_isa_t isa>
+bool jitGatherKernelBase<isa>::isSameParams(const jGatherConfParams& jcp) {
+    return beforeAxisSize == jcp.beforeAxisSize && specIdxSize == jcp.specIdxSize && batchDims == jcp.batchDims &&
+            reverseIndexing == jcp.reverseIndexing && afterAxisSize == jcp.afterAxisSize && dynamicShapes && jcp.dynamicShapes &&
```

Comment:
The reverseIndexing depends on operation version and runtime parameter, i.e. it is constant and should not be checked here.

Reply:
Thank you for this clarification. I will fix this.